### ISSUE TITLE
Bookmark crash in BookmarkEditActivity

### ISSUE THREAD

**Issue by GuyInTheShadows** [bromite/bromite#172]
**Bromite version**

Version: `70.0.3538.87`
Arch: `arm64`
Android version: `8.1`

**Is the bug reproducible with latest version?**
Yes.

**Can the bug be reproduced with corresponding [Chromium](https://github.com/bromite/chromium/releases) version?**
No.

**Is the bug a crash?**
Yes.

Logcat:
```
--------- beginning of main
11-04 11:57:25.663  1734  1951 E LightsService: Light requested not available on this device. 2
11-04 11:57:27.202  1006  1098 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
11-04 11:57:27.207  1734  2080 E LightsService: Light requested not available on this device. 2
11-04 11:57:27.356 22031 22036 I zygote64: Do full code cache collection, code=118KB, data=90KB
11-04 11:57:27.357 22031 22036 I zygote64: After code cache collection, code=115KB, data=66KB
11-04 11:57:27.377 22031 22031 W cr_Autocomplete: stopping autocomplete.
11-04 11:57:27.917  1006  1098 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
--------- beginning of system
11-04 11:57:28.021  1734 22745 I ActivityManager: START u0 {cmp=org.bromite.bromite/org.chromium.chrome.browser.bookmarks.BookmarkEditActivity (has extras)} from uid 10065
11-04 11:57:28.103 22031 22048 W chromium: [WARNING:library_prefetcher.cc(300)] Incorrect code ordering
11-04 11:57:28.138 22031 22031 I TextInputLayout: EditText added is not a TextInputEditText. Please switch to using that class instead.
11-04 11:57:28.144 22031 22031 I TextInputLayout: EditText added is not a TextInputEditText. Please switch to using that class instead.
11-04 11:57:28.154 22031 22031 D AndroidRuntime: Shutting down VM
--------- beginning of crash
11-04 11:57:28.154 22031 22031 E AndroidRuntime: FATAL EXCEPTION: main
11-04 11:57:28.154 22031 22031 E AndroidRuntime: Process: org.bromite.bromite, PID: 22031
11-04 11:57:28.154 22031 22031 E AndroidRuntime: java.lang.AssertionError
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at SS.a(PG:296)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at SS.onActivityStarted(PG:283)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.Application.dispatchActivityStarted(Application.java:207)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.Activity.onStart(Activity.java:1245)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at fc.onStart(PG:564)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at og.onStart(PG:177)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1334)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.Activity.performStart(Activity.java:7029)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2741)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.ActivityThread.-wrap11(Unknown Source:0)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6494)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
11-04 11:57:28.154 22031 22031 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
11-04 11:57:28.158  1734 16461 W ActivityManager:   Force finishing activity org.bromite.bromite/org.chromium.chrome.browser.bookmarks.BookmarkEditActivity
11-04 11:57:28.160  1734 16461 W ActivityManager:   Force finishing activity org.bromite.bromite/com.google.android.apps.chrome.Main
11-04 11:57:28.163  1734  1891 I ActivityManager: Showing crash dialog for package org.bromite.bromite u0
11-04 11:57:28.198  1734  3543 I OpenGLRenderer: Initialized EGL, version 1.4
11-04 11:57:28.198  1734  3543 D OpenGLRenderer: Swap behavior 2
11-04 11:57:28.368   978  1562 D NativeSensors: CapSensor:  mPendingEvent.distance=2.000000
11-04 11:57:28.660  1734  1890 W ActivityManager: Activity pause timeout for ActivityRecord{8aca945 u0 org.bromite.bromite/org.chromium.chrome.browser.bookmarks.BookmarkEditActivity t213 f}
11-04 11:57:29.277  2866  2971 W OpenGLRenderer: Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
11-04 11:57:29.277  2866  2971 I chatty  : uid=10016(com.android.launcher) RenderThread identical 1 line
11-04 11:57:29.277  2866  2971 W OpenGLRenderer: Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
11-04 11:57:29.302  1006  1098 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
11-04 11:57:29.545  1734 16461 W ActivityManager:   Force finishing activity org.bromite.bromite/org.chromium.chrome.browser.bookmarks.BookmarkEditActivity
11-04 11:57:29.546  1734 16461 W ActivityManager:   Force finishing activity org.bromite.bromite/com.google.android.apps.chrome.Main
11-04 11:57:29.563  1734 16461 I ActivityManager: Killing 22031:org.bromite.bromite/u0a65 (adj 900): crash
11-04 11:57:29.536  1006  1098 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
11-04 11:57:29.564  1734  1892 W zygote64: kill(-22031, 9) failed: No such process
11-04 11:57:29.567 22061 22061 I cr_ChildProcessService: Destroying ChildProcessService pid=22061
11-04 11:57:29.567 22093 22093 I cr_ChildProcessService: Destroying ChildProcessService pid=22093
11-04 11:57:29.571 22093 22120 I chromium: [INFO:frame_sink_manager_impl.cc(62)] Check failed: sink_map_.empty(). 
11-04 11:57:29.571 22093 22120 I chromium: [INFO:frame_sink_manager_impl.cc(63)] Check failed: registered_sources_.empty(). 
11-04 11:57:29.577  1734  3712 I ActivityManager: Killing 22061:org.bromite.bromite:sandboxed_process0/u0a65i562 (adj 0): isolated not needed
11-04 11:57:29.602  1734  1892 W zygote64: kill(-22031, 9) failed: No such process
11-04 11:57:29.624  1734  2385 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ LISTEN id=696, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&FOREGROUND] ], android.os.BinderProxy@af1d74d)
11-04 11:57:29.625  1734  7065 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ LISTEN id=693, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&FOREGROUND] ], android.os.BinderProxy@b205202)
11-04 11:57:29.625  1734  1949 W ActivityManager: setHasOverlayUi called on unknown pid: 22031
11-04 11:57:29.627  1734  2325 D ConnectivityService: ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ LISTEN id=695, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED&FOREGROUND] ], android.os.BinderProxy@54bb713)
11-04 11:57:29.634  1734  1915 D KernelUidCpuTimeReader: Removing uids 99562-99562
11-04 11:57:29.614  1734  1892 I chatty  : uid=1000(system) ActivityManager identical 1 line
11-04 11:57:29.640  1734  1892 W zygote64: kill(-22031, 9) failed: No such process
11-04 11:57:29.640  1734  1892 I zygote64: Successfully killed process cgroup uid 10065 pid 22031 in 75ms
11-04 11:57:29.640  1734  1892 W zygote64: kill(-22061, 9) failed: No such process
11-04 11:57:29.640  1734  1892 I zygote64: Successfully killed process cgroup uid 99562 pid 22061 in 0ms
11-04 11:57:29.650  1734  8488 I ActivityManager: Process org.bromite.bromite:privileged_process0 (pid 22093) has died: cch  CEM 
11-04 11:57:29.650  1734  1892 W zygote64: kill(-22093, 9) failed: No such process
11-04 11:57:29.679  1734  2039 W AppOps  : Finishing op nesting under-run: uid 1000 pkg android code 24 time=0 duration=0 nesting=0
11-04 11:57:29.691  1734  1892 W zygote64: kill(-22093, 9) failed: No such process
11-04 11:57:29.691  1734  1892 I zygote64: Successfully killed process cgroup uid 10065 pid 22093 in 41ms
```

**Describe the bug**
When you try to delete bookmark from the hamburger menu, Bromite crashes.

**To Reproduce**

1. Bookmark a website. For this example we use `github.com`.
2. Go to bookmarked website, `github.com`.
3. Open hamburger menu and press blue star to remove bookmark.
4. Bromite crashes and bookmark isn't removed.

**Expected behavior**
It should remove the bookmark and not to crash.

**Screenshots**
Video of the bug:
https://linx.li/59a99pt2.mp4

**Comment by csagan5** [comment]
@GuyInTheShadows please report the slowness problem in a separate issue, we cannot have one issue for two problems.

I cannot reproduce this crash by deleting bookmarks. Can you verify how to reproduce it? Did you import some bookmark before deleting or...?

**Comment by csagan5** [comment]
@GuyInTheShadows I can reproduce it by clicking "edit bookmark" after adding it. Can you confirm that you have previously clicked to edit the bookmark and that's why it crashes?

In your crash log I see `BookmarkEditActivity` and that's why I tried editing instead and that way I can see the crash, no problems with deletion per se.

**Comment by GuyInTheShadows** [comment]
> I can reproduce it by clicking "edit bookmark" after adding it. 

I just tested "edit bookmark" and it happens for me too, Bromite crashes.

> Can you confirm that you have previously clicked to edit the bookmark and that's why it crashes?

No. It happens for me in both ways. Either by deleting bookmark **or** by editing bookmark.

**Comment by csagan5** [comment]
Also in this case, I believe this issue was not tested with Chromium:
> Can the bug be reproduced with corresponding Chromium version?
> No.
> 

The bug can be reproduced with Chromium.

**Comment by csagan5** [comment]
This has been fixed in next release `70.0.3538.95` (not yet released).
